lib/puppet/provider/zfs_share/solaris.rb

```diff
@@ -0,0 +1,223 @@
+Puppet::Type.type(:zfs_share).provide(:solaris) do
```

Comment:
I reviewed from bottom up and after a full review...

You should be using getter and setters for all parameters as opposed to abusing `exists?` and `create`.
Needs to stop using the `zfs set share=` pattern.